File: Java/LeetCode/Convert Date to Binary.java
// https://leetcode.com/problems/convert-date-to-binary/

class Solution {

    public String convertDateToBinary(String date) {
        String[] parts = date.split("-");

        for (int i = 0; i < parts.length; ++i) {
            parts[i] = Integer.toBinaryString( Integer.parseInt(parts[i]) );
        }

        return String.join("-", parts);
    }

    // Alternative
    /*
    public String convertDateToBinary(String date) {
        return Stream.of(date.split("-")).map(part -> Integer.toBinaryString(Integer.parseInt(part))).collect(Collectors.joining("-"));
    }
    */

}

File: Kotlin/LeetCode/Convert Date to Binary.kt
// https://leetcode.com/problems/convert-date-to-binary/

class Solution {
    
    fun convertDateToBinary(date: String) = date.split("-").map { it.toInt().toString(2) }.joinToString("-")

    // Alternative
    /*
    fun convertDateToBinary(date: String): String {
        var parts = date.split("-").toMutableList()

        for (i in parts.indices) {
            parts[i] = parts[i].toInt().toString(2)
        }

        return parts.joinToString("-")
    }
    */

}
